services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/system_monitor/_disk_usage.py

```diff
@@ -17,41 +21,177 @@
 from ..mounted_fs import MountedVolumes
 from ..notifications import publish_disk_usage
 
+_NODE_FILE_SYSTEM_PATH: Final[Path] = Path("/")
+
+
 _logger = logging.getLogger(__name__)
 
 
+class MountPathCategory(StrAutoEnum):
+    HOST = auto()
+    STATES_VOLUMES = auto()
+    INPUTS_VOLUMES = auto()
+    OUTPUTS_VOLUMES = auto()
+
+
+_SUPPORTED_ITEMS: Final[set[str]] = {
+    MountPathCategory.HOST,
+    MountPathCategory.STATES_VOLUMES,
+}
+
+
 async def get_usage(path: Path) -> DiskUsage:
     usage = await asyncio.get_event_loop().run_in_executor(
         None, psutil.disk_usage, f"{path}"
     )
     return DiskUsage.from_ps_util_disk_usage(usage)
 
 
+def get_relative_path(path: Path, dy_volumes_mount_dir: Path) -> Path:
+    try:
+        return path.relative_to(dy_volumes_mount_dir)
+    except ValueError:
+        return path
+
+
+def _get_normalized_folder_name(path: Path) -> str:
+    return f"{path}".replace("/", "_")
+
+
+def _have_common_entries(a: set[str], b: set[str]) -> bool:
+    return len(a & b) > 0
+
+
 @dataclass
 class DiskUsageMonitor:
     app: FastAPI
     user_id: UserID
     node_id: NodeID
     interval: timedelta
-    monitored_paths: list[Path]
+    monitored_paths: dict[MountPathCategory, set[Path]]
+
+    dy_volumes_mount_dir: Path
     _monitor_task: asyncio.Task | None = None
-    _last_usage: dict[Path, DiskUsage] = field(default_factory=dict)
+    _incoming_overwrite_usage: dict[str, DiskUsage] = field(default_factory=dict)
+    _last_usage: dict[str, DiskUsage] = field(default_factory=dict)
+
+    @cached_property
+    def _flat_monitored_paths(self) -> set[Path]:
+        return set.union(*self.monitored_paths.values())
+
+    @cached_property
+    def _normalized_monitored_paths(self) -> dict[MountPathCategory, set[str]]:
+        return {
+            k: {
+                _get_normalized_folder_name(
+                    get_relative_path(p, self.dy_volumes_mount_dir)
+                )
+                for p in paths
+            }
+            for k, paths in self.monitored_paths.items()
+        }
+
+    async def _get_measured_disk_usage(self) -> list[DiskUsage]:
+        return await logged_gather(
+            *[
+                get_usage(monitored_path)
+                for monitored_path in self._flat_monitored_paths
+            ]
+        )
 
-    async def _publish_disk_usage(self, usage: dict[Path, DiskUsage]):
+    def _get_normalized_disk_usage(
```

Comment:
so what is a normalized disk usage?

Reply:
naming was wron, renamed and added docstring to function